upup/pkg/fi/cloudup/awstasks/route.go

```diff
@@ -32,9 +32,13 @@ type Route struct {
 	Name *string
 
 	RouteTable      *RouteTable
-	InternetGateway *InternetGateway
 	Instance        *Instance
 	CIDR            *string
+
+	// Either an InternetGateway or a NAT Gateway
+	// MUST be provided.
+	InternetGateway *InternetGateway
+	NatGateway      *NatGateway
 }
 
 func (e *Route) Find(c *fi.Context) (*Route, error) {
```

Comment:
Again .. `Find` What the heck are we doin here.  @justinsb we need to talk about function names 🙊 No criticism, lets make this stuff AWESOME.